contracts/optimism/L1ERC20TokenBridge.sol

```diff
@@ -9,37 +9,56 @@ import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol
 
 import {IL1ERC20Bridge} from "./interfaces/IL1ERC20Bridge.sol";
 import {IL2ERC20Bridge} from "./interfaces/IL2ERC20Bridge.sol";
-
+import {IERC20Wrapper} from "../token/interfaces/IERC20Wrapper.sol";
 import {BridgingManager} from "../BridgingManager.sol";
-import {BridgeableTokens} from "../BridgeableTokens.sol";
+import {RebasableAndNonRebasableTokens} from "./RebasableAndNonRebasableTokens.sol";
 import {CrossDomainEnabled} from "./CrossDomainEnabled.sol";
+import {DepositDataCodec} from "./DepositDataCodec.sol";
+import {IERC20WstETH} from "../token/interfaces/IERC20WstETH.sol";
 
-/// @author psirex
+/// @author psirex, kovalgek
 /// @notice The L1 ERC20 token bridge locks bridged tokens on the L1 side, sends deposit messages
 ///     on the L2 side, and finalizes token withdrawals from L2. Additionally, adds the methods for
 ///     bridging management: enabling and disabling withdrawals/deposits
-contract L1ERC20TokenBridge is
+abstract contract L1ERC20TokenBridge is
```

Comment:
```suggestion
abstract contract L1WrappedRebasableTokensBridge is
```
or at least introduce notions of multiple tokens and rebasability
although I struggle to propose a nice name